src/coreclr/tools/Common/TypeSystem/Interop/InteropStateManager.cs

```diff
@@ -478,12 +478,12 @@ public PInvokeLazyFixupFieldHashtable(DefType owningType)
             }
         }
 
-        private class PInvokeCalliHashtable : LockFreeReaderHashtable<MethodSignature, CalliMarshallingMethodThunk>
+        private class PInvokeCalliHashtable : LockFreeReaderHashtable<(MethodSignature Signature, bool UseRuntimeMarshalling), CalliMarshallingMethodThunk>
         {
             private readonly InteropStateManager _interopStateManager;
             private readonly TypeDesc _owningType;
 
-            protected override int GetKeyHashCode(MethodSignature key)
+            protected override int GetKeyHashCode((MethodSignature Signature, bool UseRuntimeMarshalling) key)
             {
                 return key.GetHashCode();
```

Comment:
```suggestion
                return key.Signature.GetHashCode();
```

The hashcode computed from the key needs to be identical to the hashcode computed from value in the method below, or bad races will happen.